sql/service/src/main/java/org/apache/spark/sql/service/cli/TypeDescriptor.java

```diff
@@ -77,10 +80,27 @@ public TypeDescriptor(String typeName) {
 
   public Type getTypeInfo(String typeName) {
     Type type = cachedPrimitiveTypeInfo.getOrDefault(typeName, Type.STRING_TYPE);
-    if (typeName.equalsIgnoreCase("decimal")) {
```

Comment:
This will only be used for QualifiedTypes, so DECIMAL and VARCHAR?
You could make this private, and get rid of `cachedPrimitiveTypeInfo` (just get Type.DECIMAL_TYPE).

Is parsing the length qualifier for VARCHAR needed as well? (don't think Spark will respect that anywhere...)

Reply:
> This will only be used for QualifiedTypes, so DECIMAL and VARCHAR?
> You could make this private, and get rid of `cachedPrimitiveTypeInfo` (just get Type.DECIMAL_TYPE).
> 
> Is parsing the length qualifier for VARCHAR needed as well? (don't think Spark will respect that anywhere...)

Want to replace blow method with simple way:
https://github.com/apache/hive/blob/47b22ce1c658fa1b970c3e2733871948e39afe32/serde/src/java/org/apache/hadoop/hive/serde2/typeinfo/TypeInfoFactory.java#L110-L127

Since we only need to handle type spark supported. According to here we only need to handle DECIMAL.
https://github.com/apache/hive/blob/47b22ce1c658fa1b970c3e2733871948e39afe32/serde/src/java/org/apache/hadoop/hive/serde2/typeinfo/TypeInfoFactory.java#L149-L174